src/lava/magma/compiler/compiler.py

```diff
@@ -194,7 +194,16 @@ def _select_proc_models(
             models: ty.List[ty.Type[AbstractProcessModel]],
             run_cfg: RunConfig) -> ty.Type[AbstractProcessModel]:
         """Selects a ProcessModel from list of provided models given RunCfg."""
-        return run_cfg.select(proc, models)
+        selected_proc_model = run_cfg.select(proc, models)
+        err_msg = f"RunConfig {run_cfg.__class__.__qualname__}.select() must " \
+            f"return a sub-class of AbstractProcessModel. Got" \
+            f" {type(selected_proc_model)} instead."
+        if not isinstance(selected_proc_model, type):
```

Comment:
What does this do, out of curiosity?

Reply:
Checks if `selected_proc_model` is a class and not a primitive data type like `str` or `int`